### PR TITLE
reference/transactions: update pessimistic doc

### PR DESCRIPTION
**Pull request by juliezhang1112**
### What is changed, added or deleted? 
- Update `transaction-pessimistic.md`

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1826

### Which version does your change affect? 
dev, v3.0

### CODE REVIEW

File: dev/reference/transactions/transaction-pessimistic.md [hunk 1]
```diff
@@ -30,56 +26,37 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
+- Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to adopt the pessimistic transaction mode and execute the explicit transactions.
 
-    ```
-    [pessimistic-txn]
-    enable = true
-    default = true
-    ```
-
-If the pessimistic transaction mode is enabled in the configuration file by default, use one of the following methods to adopt the optimistic transaction mode for the transaction:
+After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is enabled by default; but you can use the following two methods to adopt the optimistic transaction mode for the transaction:
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is adopted by default; but you can use either of the following two methods to adopt the optimistic transaction mode for the transaction:
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 2]
```diff
@@ -30,56 +26,37 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
 - Execute the `set @@tidb_txn_mode = 'optimistic';` statement to allow all the transactions processed in this session to be in the optimistic transaction mode.
 
-## Enablement priority
-
-The three methods to enable the transaction mode are ordered from highest priority to lowest as follows:
-
-- `BEGIN PESSIMISTIC;` or `BEGIN OPTIMISTIC;`.
-
-- The session variable `tidb_txn_mode`.
+The `BEGIN PESSIMISTIC;` and `BEGIN OPTIMISTIC;` statements take precedence over the `tidb_txn_mode` system variable. Transactions that are opened with these two statements ignore system variables.
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
The `BEGIN PESSIMISTIC;` and `BEGIN OPTIMISTIC;` statements take precedence over the `tidb_txn_mode` system variable. Transactions that are opened with these two statements will ignore system variables.
```

Comment (dcalvin) [reply dev/reference/transactions/transaction-pessimistic.md]: Here I suggest we add a "will" to denote the future behavior based on judgment.

File: dev/reference/transactions/transaction-pessimistic.md [hunk 3]
```diff
@@ -30,56 +26,37 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
 
-- The `default` configuration item in the configuration file. If you use a regular `BEGIN` statement and set the value of `tidb_txn_mode` to an empty string, then you can use `default` to determine whether to enable the pessimistic or optimistic transaction mode.
+To disable the pessimistic transaction mode, you can modify the configuration file and add `enable = false` to the `[pessimistic-txn]` category.
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
To disable the pessimistic transaction mode, modify the configuration file and add `enable = false` to the `[pessimistic-txn]` category.
```

File: dev/reference/transactions/transaction-pessimistic.md [hunk 4]
```diff
@@ -30,56 +26,37 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
 
-When `enable` is set to `true`, the default transaction mode in TiDB is still optimistic. To enable the pessimistic transaction mode, choose any of the following methods that suits your needs:
+To enable the pessimistic transaction mode, choose any of the following three methods that suits your needs:
 
 - Use the `BEGIN PESSIMISTIC;` statement to start the transaction in the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
 
 - Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the explicit transactions (namely non-autocommit transactions) processed in this session to be in the pessimistic transaction mode.
 
-- Enable the pessimistic transaction mode in the configuration file. This allows all transactions (except auto-committed single-statement ones) to adopt the pessimistic transaction mode.
+- Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to adopt the pessimistic transaction mode and execute the explicit transactions.
```
Comment (dcalvin) [inline dev/reference/transactions/transaction-pessimistic.md]:
```suggestion
- Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to adopt the pessimistic transaction mode and execute explicit transactions.
```

Comment (coocood) [reply dev/reference/transactions/transaction-pessimistic.md]: Should be `to` or `when` instead of `and` in `and execute the explicit transactions`

Comment (coocood) [reply dev/reference/transactions/transaction-pessimistic.md]: `apply` is better than `adopt`.

**Comment by juliezhang1112** [comment]
PTAL @dcalvin @coocood

File: dev/reference/transactions/transaction-pessimistic.md [hunk 5]
```diff
@@ -30,56 +26,37 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
-- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the explicit transactions (namely non-autocommit transactions) processed in this session to be in the pessimistic transaction mode.
-
-- Enable the pessimistic transaction mode in the configuration file. This allows all transactions (except auto-committed single-statement ones) to adopt the pessimistic transaction mode.
-
-    ```
-    [pessimistic-txn]
-    enable = true
-    default = true
-    ```
+To enable the pessimistic transaction mode, choose any of the following three methods that suits your needs:
```
Comment (coocood) [inline dev/reference/transactions/transaction-pessimistic.md]: `enable` -> `apply`

**Comment by coocood** [comment]
LGTM

**Comment by juliezhang1112** [comment]
PTAL again, thanks! @dcalvin

**Pull request merged**